src/app/model/ChessEngine.ts

```diff
@@ -68,7 +70,78 @@ export class ChessEngine implements ChessModel {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
 
+  private returnWirtualProposedBoardAfterMove(chessboard: Chessboard, squareFrom: Square, squareTo: Square): Chessboard {
+    const wirtualPorposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = wirtualPorposedChessboard;
+
+    const movedPiece = chessboard.onPositionPiece(squareFrom);
+    const positionFrom = `${squareFrom.column}${squareFrom.row}`;
+    const positionTo = `${squareTo.column}${squareTo.row}`;
+    delete proposedSquaresWithPieces[positionFrom];
+    proposedSquaresWithPieces[positionTo] = movedPiece as Piece;
+
+    return wirtualPorposedChessboard;
+  }
+
+  public static kingPosition(chessboard: Chessboard, kingSide: Side): Square | undefined {
+    const { squaresWithPiece: squaresWithPieces } = chessboard;
+    const kingPositionKey = Object.keys(squaresWithPieces).find((key) => {
+      const isKingName = squaresWithPieces[key].name === 'King';
+      const isPlayerSide = squaresWithPieces[key].side === kingSide;
+      return isKingName && isPlayerSide;
+    });
+    const kingPosition = kingPositionKey
+      ? {
+          column: kingPositionKey[0],
+          row: Number(kingPositionKey[1]) as Row,
+        }
+      : undefined;
+
+    return kingPosition;
+  }
```

Comment:
Proponuje cos takiego, a najlepiej jako metoda instancji Chessboard :) wtedy tylko `chessboard.kingPosition(forSide: Side): Square`
```suggestion
  private static kingPosition(chessboard: Chessboard, kingSide: Side): Square {
    const { squaresWithPiece: squaresWithPieces } = chessboard;
    const kingPositionKey = Object.keys(squaresWithPieces)
      .find((key) => squaresWithPieces[key].equals(new King(kingSide)));
    if (!kingPositionKey) {
      throw new Error(`Chess board is invalid, because there is no King for side ${kingSide}!`);
    }
    return {
      column: kingPositionKey[0],
      row: Number(kingPositionKey[1]) as Row,
    };
  }
```